### PR TITLE
[SDK-5158] Add docs about React location library fro Android

### PR DESCRIPTION
**Pull request by chshapiro**
### Summary

My PR...

### Related PRs, issues, or features (optional)

- N/A

### Feature release date (optional)

- N/A

### Contributor checklist

- [x] I confirm that my PR meets the following:
    - I signed the [Contribution License Agreement (CLA)](https://www.braze.com/docs/cla).
    - My style and voice follows the [Braze Style Guide](https://docs.google.com/document/u/2/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub).
    - My content contains correct spelling and grammar.
    - All links are working correctly.
    - If I renamed or moved a file or directory, I set up [URL redirects](https://www.braze.com/docs/contributing/content_management/redirecting_urls) for each file.
    - If I updated or replaced an image, I did not remove the original image file from the repository. (For more information, see [Updating an image](https://www.braze.com/docs/contributing/content_management/images/#updating-an-image)).
    - If my PR is related to a paid SKU, third party, SMS, AI, or privacy, I have received written approval from Braze Legal.

### Submitting for review

If your PR meets the above requirements, select **Ready for review**, then add a reviewer:
- **Non-Braze contributors:** Add `braze-inc/docs-team` as the reviewer.
- **Braze employees:** Have any relevant subject matter experts (like engineers or product managers) review your work first, then add the [tech writer assigned to your specific vertical](https://confluence.braze.com/pages/viewpage.action?pageId=75039896). If you're not sure which tech writer to add, you can add `braze-inc/docs-team` instead. 

Thanks for contributing! We look forward to reading your work.

### CODE REVIEW

**Review by vanessaland — APPROVED** [review]

**Comment by chshapiro** [comment]
@internetisaiah ready to merge!

File: _docs/_developer_guide/platform_integration_guides/react_native/react_sdk_setup.md [hunk 1]
```diff
@@ -175,6 +175,8 @@ Add the required permissions to your `AndroidManifest.xml` file:
 <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
 ```
 
+Starting in version 12.2.0, set `importBrazeLocationLibrary=true` in your `gradle.properties` file to automatically pull in the android-sdk-location library
```
Comment (internetisaiah) [inline _docs/_developer_guide/platform_integration_guides/react_native/react_sdk_setup.md:178]:
```suggestion
{% alert tip %}
On Braze SDK version 12.2.0 or later, you can automatically pull in the android-sdk-location library, by setting `importBrazeLocationLibrary=true` in your `gradle.properties` file .
{% endalert %}
```

File: _docs/_developer_guide/platform_integration_guides/react_native/react_sdk_setup.md [hunk 2]
```diff
@@ -175,6 +175,10 @@ Add the required permissions to your `AndroidManifest.xml` file:
 <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
 ```
 
+{% alert tip %}
+On Braze SDK version 12.2.0 or later, you can automatically pull in the android-sdk-location library, by setting `importBrazeLocationLibrary=true` in your `gradle.properties` file .
```
Comment (internetisaiah) [inline _docs/_developer_guide/platform_integration_guides/react_native/react_sdk_setup.md:179]:
```suggestion
On Braze SDK version 12.2.0 or later, you can automatically pull in the android-sdk-location library by setting `importBrazeLocationLibrary=true` in your `gradle.properties` file .
```

**Review by internetisaiah — APPROVED** [review]
@chshapiro  LGTM! Friendly reminder to fill out the PR description with a reason for making the change so we can have historical info on changes. That being said, I'll still merge your PR now.

Thanks for contributing!

**Pull request merged**